core/controllers/reader.py

```diff
@@ -705,31 +705,34 @@ def get(self, exploration_id):
         except Exception:
             raise self.PageNotFoundException
 
-        auto_recommended_exp_ids = []
+        system_recommended_exp_ids = []
+        next_exp_id = None
 
         if collection_id:
             if self.user_id:
-                auto_recommended_exp_ids = (
-                    collection_services.get_next_exploration_ids_to_complete_by_user(  # pylint: disable=line-too-long
+                next_exp_id = (
+                    collection_services.get_next_exploration_id_to_complete_by_user(  # pylint: disable=line-too-long
                         self.user_id, collection_id))
             else:
                 collection = collection_services.get_collection_by_id(
                     collection_id)
-                auto_recommended_exp_ids = (
-                    collection.get_next_exploration_ids_in_sequence(
+                next_exp_id = (
+                    collection.get_next_exploration_id_in_sequence(
                         exploration_id))
         elif include_system_recommendations:
             system_chosen_exp_ids = (
                 recommendations_services.get_exploration_recommendations(
                     exploration_id))
             filtered_exp_ids = list(
                 set(system_chosen_exp_ids) - set(author_recommended_exp_ids))
-            auto_recommended_exp_ids = random.sample(
+            system_recommended_exp_ids = random.sample(
                 filtered_exp_ids,
                 min(MAX_SYSTEM_RECOMMENDATIONS, len(filtered_exp_ids)))
 
         recommended_exp_ids = set(
-            author_recommended_exp_ids + auto_recommended_exp_ids)
+            author_recommended_exp_ids + system_recommended_exp_ids)
+        recommended_exp_ids.add(next_exp_id)
```

Comment:
next_exp_id could be None if the elif branch is taken; is it your intention to add that to the set?

Reply:
Done, added a check for it.